### ISSUE TITLE
Minimize Environment Variables

### ISSUE THREAD

**Issue by crwilcox** [GoogleCloudPlatform/google-cloud-node#2851]
During a summit, @alexander-fenster mentioned that we have too many different Environment variables.

- Google_Application_default_credentials
- gcloud_project_id
- etc.

**Comment by JustinBeckwith** [comment]
As far as I can tell, we only respect 3 environment variables:
- `GOOGLE_APPLICATION_CREDENTIALS` - This provides the path to a keyfile with the rest of the good stuff in it.  
- `GCLOUD_PROJECT` - This was the original way for a user to express the project Id via environment variable.  It was a weird name. So I think we moved to...
- `GOOGLE_CLOUD_PROJECT` - This is the new way of specifying the project Id via environment variable.  This is preferred.

I believe this is consistent with Python (calling @theacodes and @crwilcox):
http://google-auth.readthedocs.io/en/latest/reference/google.auth.environment_vars.html

TL;DR - I actually think we're fine here.  @alexander-fenster am I missing something?

**Comment by theacodes** [comment]
This is correct

**Comment by JustinBeckwith** [comment]
I'm going to go ahead and close this out for now.  If folks want to see a change here, please do let me know!
